tags.go

```diff
@@ -0,0 +1,42 @@
+package distribution
+
+// Tag is a named reference to a repository object. Target generally refers to
+// a manifest.
+type Tag interface {
+	Blob // io access to serialized content.
+
+	// Name returns the fully qualified name of the object. For example, if
+	// the object is in repository "stevvooe/test" and has tag "latest", the
+	// name will be "stevvooe/test:latest".
+	Name() string
+
+	// Description returns a human-readable description of the tag.
+	Description() string
+
+	// Target identifies the blob targeted by this tag.
+	Target() Descriptor
+}
+
+// TagService provides access to information about tagged objects.
+type TagService interface {
+	All() (Tag, error)
+
+	// Get retrieves the digest for the provided tag.
+	Get(name string) (Tag, error)
```

Comment:
These tags should be able to be fetched by digest as well.  If tags are being used as the method of trust by an external trust system, it will never fetch content by name but only content address.  Since signatures are on the tags, an external system will consider the tag to be verified and use its content-address to start the trusted lookup.